### DISCUSSION TITLE
Android

### DISCUSSION THREAD

**Discussion by InnCreator** [ihhub/fheroes2#4974] [Ideas]
Hi. I have 3 years commerce android developer. And I can try port you project on android. I have several question. Can I create kotlin project in android studio.  And how work with game controls? maybe override some interface

**Comment by ihhub** [comment]
Hi @InnCreator !

You're welcome to use any widely used programming language if it's OS independent and it's easy to implement into our automated CI/CD pipeline. The game engine is based on SDL  library which is responsible for all controls. In other words, we need a wrapper application over existing C++ application which starts it. No other manipulations (as of now) are required.
